lib/filterable_sortable.rb

```diff
@@ -10,18 +10,43 @@ module FilterableSortable
       if filter[:search]
         search(filter[:search])
       elsif filter[:custom] && filter[:custom] != 'all'
-        self.send(filter[:custom].to_sym) if self.methods.include?(filter[:custom].to_sym)
+        if methods.include?(filter[:custom].to_sym) &&
+          !(filter[:custom] =~ /delete_|update_|create|drop_|destroy_|\!/)
+          readonly.send(filter[:custom].to_sym)
+        end
       end
     }
 
     scope :search, lambda { |term|
-      conditions = self.columns_hash.collect{|k,v| k unless v.type.in?([:datetime, :time, :date])}.compact.collect{|f| "#{self.table_name}.#{f} like '%#{term}%'"}.join(' OR ')
+      conditions =
+        columns_hash.
+        map { |k, v| k unless v.type.in?([:datetime, :time, :date]) }.
+        compact.
+        map do |f|
+          "#{table_name}.#{f} like " \
+          "#{ActiveRecord::Base.sanitize('%#{term}%')}"
+        end.
+        join(' OR ')
       where(conditions)
     }
 
     scope :ordered, lambda { |ordered|
-      ordered[:field] = "#{self.table_name}.#{ordered[:field]}" unless ordered[:field].match(/\./)
-      order("#{ordered[:field]} #{ordered[:direction]}") if ordered
+      ordered[:field] =
+        if ordered[:field] =~ /\./
+          ActiveRecord::Base.sanitize "#{table_name}.#{ordered[:field]}"
+        else
+          ActiveRecord::Base.sanitize ordered[:field]
+        end
+      if ordered
+        case ordered[:direction].to_s.capitalize
+        when "ASC"
+          order(ordered[:field] => :ASC)
```

Comment:
It seems `order(ordered[:field] => :ASC)` and `order(ordered[:field])` gets same result